nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -85,9 +162,9 @@ impl<'a> Planner<'a> {
         let mut sleds_ineligible_for_services = BTreeSet::new();
 
         for (sled_id, sled_info) in &self.policy.sleds {
-            // Decommissioned sleds don't get any services. (This is an
-            // explicit match so that when more states are added, this fails to
-            // compile.)
+            // Decommissioned and expunged sleds don't get any services. (These
+            // are explicit matches so that when more states are added, this
+            // fails to compile.)
```

Comment:
Similarly I wonder if this could just look at "expunged" now.  It feels like only the policy should determine how we plan resources for the sled.